### PR TITLE
New option for Modification allowed settings in Registration

### PR DESCRIPTION
**Pull request by vasantvohra**
-  Add a new `Until approved` option for a registration form's "Modification allowed" settings.

### CODE REVIEW

File: indico/modules/events/registration/models/forms.py
Comment (ThiefMaster) [inline indico/modules/events/registration/models/forms.py:38]: You can't re-number the enum like this - doing so would require updating the value for ALL existing regforms. BTW, regardless of this, you need an alembic revision to recreate the CHECK constraint on the column where this enum is used to allow the new value.

File: CHANGES.rst
Comment (ThiefMaster) [inline CHANGES.rst:12]:
```suggestion
- Add a new "Until approved" option for a registration form's "Modification allowed"
  setting (:pr:`4740`, thanks :user:`vasantvohra`)
```

File: indico/modules/events/registration/models/forms.py
Comment (ThiefMaster) [inline indico/modules/events/registration/models/forms.py:411]: there's `!=` ;)

Also, I think this setting should explicitly check for the `pending` state.

File: indico/modules/events/registration/templates/display/registration_summary.html
Comment (ThiefMaster) [inline indico/modules/events/registration/templates/display/registration_summary.html:51]: I don't think adding this check here is correct. We have cases that disable and show detailed tooltips within this if block exactly in case the registration cannot be modified.

**Review by ThiefMaster — CHANGES_REQUESTED** [review]

File: indico/migrations/versions/20201209_2010_e4fb983dc64c_modification_mode_enum.py
Comment (ThiefMaster) [inline indico/migrations/versions/20201209_2010_e4fb983dc64c_modification_mode_enum.py:29]: Too much indentation 👾 

```suggestion
def upgrade():
    op.execute('''
        ALTER TABLE "event_registration"."forms" DROP CONSTRAINT "ck_forms_valid_enum_modification_mode";
        ALTER TABLE "event_registration"."forms" ADD CONSTRAINT "ck_forms_valid_enum_modification_mode" CHECK ((modification_mode = ANY (ARRAY[0, 1, 2, 3, 4])));
    ''')


def downgrade():
    op.execute('''
        UPDATE "event_registration"."forms" SET modification_mode = 3 WHERE modification_mode = 4;
        ALTER TABLE "event_registration"."forms" DROP CONSTRAINT "ck_forms_valid_enum_modification_mode";
        ALTER TABLE "event_registration"."forms" ADD CONSTRAINT "ck_forms_valid_enum_modification_mode" CHECK ((modification_mode = ANY (ARRAY[0, 1, 2, 3])));
    ''')
```

File: indico/modules/events/registration/models/forms.py
Comment (ThiefMaster) [inline indico/modules/events/registration/models/forms.py:35]: hm, when discussing about this with @OmeGak recently i suggested that "While pending approval" may be a good title that's clearer. no strong opinion on this though

Comment (vasantvohra) [reply indico/modules/events/registration/models/forms.py]: Done

File: indico/migrations/versions/20201209_2010_e4fb983dc64c_modification_mode_enum.py
Comment (ThiefMaster) [inline indico/migrations/versions/20201209_2010_e4fb983dc64c_modification_mode_enum.py:1]: That's not really a useful revision title. Let's change this to something like `Add 'while pending' regform modification mode` and rename the file to `20201209_2010_e4fb983dc64c_add_while_pending_regform_modification_mode.py`

File: indico/migrations/versions/20201209_2010_e4fb983dc64c_modification_mode_enum.py
Comment (ThiefMaster) [inline indico/migrations/versions/20201209_2010_e4fb983dc64c_modification_mode_enum.py:20]:
```suggestion
        ALTER TABLE "event_registration"."forms" ADD CONSTRAINT "ck_forms_valid_enum_modification_mode" CHECK ((modification_mode = ANY (ARRAY[1, 2, 3, 4])));
```

File: indico/migrations/versions/20201209_2010_e4fb983dc64c_modification_mode_enum.py
Comment (ThiefMaster) [inline indico/migrations/versions/20201209_2010_e4fb983dc64c_modification_mode_enum.py:28]:
```suggestion
        ALTER TABLE "event_registration"."forms" ADD CONSTRAINT "ck_forms_valid_enum_modification_mode" CHECK ((modification_mode = ANY (ARRAY[1, 2, 3])));
```

**Review by ThiefMaster — CHANGES_REQUESTED** [review]

**Comment by pferreir** [comment]
Would it perhaps make more sense to call the option "untill approved"? It's more consistent with the others, IMO.

**Comment by vasantvohra** [comment]
> Would it perhaps make more sense to call the option "untill approved"? It's more consistent with the others, IMO.

I totally agree but it was @ThiefMaster or @OmeGak suggestion

**Comment by ThiefMaster** [comment]
Yeah, that's what it was initially, I had suggested "while pending" since it may be clearer since we show registrations as "pending", and I'm not sure how clear it is for people what an "approved" registration means (especially in case moderation isn't used).

File: indico/modules/events/registration/models/forms.py
Comment (vasantvohra) [reply indico/modules/events/registration/models/forms.py]: here it is @pferreir

**Comment by pferreir** [comment]
> I'm not sure how clear it is for people what an "approved" registration means

Fair enough. I actually find it less clear, since all other options specify an "until" and this one a "while". But yeah, we'll see whether people get it or not...

**Comment by ThiefMaster** [comment]
I don't have a strong opinion on this btw, so fine either way. It's right that keeping "until ..." is more consistent.

**Comment by vasantvohra** [comment]
Here you go @pferreir ..

**Comment by pferreir** [comment]
> Here you go @pferreir ..

Great, thanks!

**Review by pferreir — APPROVED** [review]
Thanks a lot for the PR!

**Comment by vasantvohra** [comment]
> Thanks a lot for the PR!

Can you squash & merge as `"I'm getting Only those with write access to this repository can merge pull requests."`

File: CHANGES.rst
Comment (ThiefMaster) [inline CHANGES.rst:12]: can you please restore this empty line? :)

**Review by ThiefMaster — APPROVED** [review]

**Comment by ThiefMaster** [comment]
ugh somehow the alembic revision disappeared from your PR... i'll forcepush to fix that

**Comment by vasantvohra** [comment]
Thanks a ton, @pferreir and @ThiefMaster,
Happy holidays will probably disturb in the new year (2021) now :100:

**Pull request merged**
